### DISCUSSION TITLE
Controller does'nt update correct value when using with useFieldArray

### DISCUSSION THREAD

**Discussion by quangnv13** [react-hook-form/react-hook-form#10166] [Q&A] [answered]
I would like to create a form array with a custom component(react-select). Then I need to use the Controller to wrap my component. I used it and I also used **remove(index)** to remove the form field in the array.
When I click the remove button the result is not the remove appreciate form field on the list. 
Example: Default I have 6 id fields. Then I click the remove button on the first field(id: 1) => and only the last field(id: 6) disappeared from the list.

Reproduce: https://stackblitz.com/edit/react-vshcjv?file=src%2FApp.js

**Comment by Sajarin-M** [comment, marked as answer]
That is because you are using the index as the key of the component. Instead, use the key provided by hook form, like:

```js
const { fields, append, remove } = useFieldArray({
  control,
  name: 'tests',
  keyName: 'key', // optional default is "id"
});

{fields.map((f, idx) => (
  <div key={f.key}>
    {/*******/}
  </div>
))}
```
See updated [example](https://stackblitz.com/edit/react-7cipbx?file=src/App.js)

**Reply by quangnv13** [reply]
Thanks, I sorted it
